### ISSUE TITLE
Client crash after restarting a scripting resource

### ISSUE THREAD

**Issue by mrgharabaghi** [altmp/altv-issues#1738]
### Description of the problem

It's better to look at `Reproduction steps`.

### Reproduction steps

1. First time `restart _vehicle-s/nitro`
3. Open map with `P` button.
4. Close map with `Esc` button.
5. Second time `restart _vehicle-s/nitro`
6. Open map with `P` button again.
7. The client will crash.

### Expected behaviour

Should not crash.

### Additional context

[CrashLogs.zip](https://github.com/altmp/altv-issues/files/10262877/CrashLogs.zip)
[client_logs.zip](https://github.com/altmp/altv-issues/files/10262898/client_logs.zip)
[nitro_resource.zip](https://github.com/altmp/altv-issues/files/10262911/nitro_resource.zip)

https://user-images.githubusercontent.com/40915667/208528714-5d48dc63-07a1-4cec-9b43-462619879ecf.mp4

### Operating system

Windows 10

### Version

release/13.37

### Reproduction tested

- [X] I confirm that I have made sure that this issue is also present on the newest dev version

**Comment by FabianTerhorst** [comment]
I think you mean client and not server

**Comment by mrgharabaghi** [comment]
> I think you mean client and not server

Sorry, my bad.

**Comment by xxshady** [comment]
@mrgharabaghi cant reproduce with steps that you provided on 14.0-dev4

**Comment by xxshady** [comment]
Me and a lot of people have had this resource restart crash on opening of pause menu, but we haven't found a way to reproduce it

**Comment by yannbcf** [comment]
@mrgharabaghi Can you still reproduce it on newest dev ?

**Comment by mrgharabaghi** [comment]
Let me try again.
@yannbcf

**Comment by mrgharabaghi** [comment]
Confirmed on `dev/14.0-dev8`.

**Comment by mrgharabaghi** [comment]
1. First time `restart temp`
2. Open map with `Esc` button.
3. The client will crash.

```toml
modules = [
    'js-module'
]

resources = [
    'temp'
]
```
[temp.zip](https://github.com/altmp/altv-issues/files/10503874/temp.zip)

**Comment by yannbcf** [comment]
Thank you, will test

**Comment by yannbcf** [comment]
I cant reproduce it, do you crash everytime you try ?

**Comment by mrgharabaghi** [comment]
Yes, everytime.

**Comment by mrgharabaghi** [comment]
After restarting the resource, spam `P` button.

**Comment by mrgharabaghi** [comment]
Restart the resource just like this video. Two times.

https://user-images.githubusercontent.com/40915667/208528714-5d48dc63-07a1-4cec-9b43-462619879ecf.mp4

**Comment by FabianTerhorst** [comment]
Its unrelated to folders.

**Comment by mrgharabaghi** [comment]
I completely regret making this (https://github.com/altmp/altv-issues/issues/641) issue. Because after adding this feature, I faced various problems.
One of them is that you must write the resource absolute path in `server.toml`.
Was there a problem with `--extra-res-folder`?

**Comment by mrgharabaghi** [comment]
What do you think about disabling the subfolder feature temporarily for testing this crash in `dev` branch?

**Comment by FabianTerhorst** [comment]
The crash is in gta and has nothing to do with the resource folders.

**Comment by xxshady** [comment]
Confirmed on 14.0-dev54

```js
import alt from 'alt-server'
import fs from 'fs'

const resource = alt.Resource.current

fs.writeFileSync(
  `${resource.path}/change-itself-client.js`, 
  'console.log("kek");\n', 
  { flag: 'a+' }
)

alt.setTimeout(() => {
  alt.restartResource(resource.name)
}, 300)
```

**Comment by C0kkie** [comment]
@mrgharabaghi  can you try on newest dev

**Comment by mrgharabaghi** [comment]
Sure

**Comment by mrgharabaghi** [comment]
Fixed, well done.

**Comment by xxshady** [comment]
@mrgharabaghi can you also try to reproduce it on release, please?

**Comment by mrgharabaghi** [comment]
> @mrgharabaghi can you also try to reproduce it on release, please?

There is no update on the release, and I think still crashed on the release branch.

**Comment by xxshady** [comment]
For some reason we cannot reproduce this crash on release (where it should not be fixed now)

**Comment by yannbcf** [comment]
I could reproduce it on release
